src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector256.cs

```diff
@@ -2338,6 +2338,67 @@ public static Vector256<sbyte> Shuffle(Vector256<sbyte> vector, Vector256<sbyte>
+        /// <remarks>Unlike Shuffle, this method delegates to the underlying hardware intrinsic without ensuring that <paramref name="indices"/> are normalized to [0, 31].
+        /// On hardware with <see cref="Avx2"/> support, but not <see cref="Avx512Vbmi.VL"/>, indices are treated as modulo 32, and if the high bit is set, the result will be set to 0 for that element.
+        /// On hardware with <see cref="Avx512Vbmi.VL"/> support, indices are treated as modulo 32.</remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [CompExactlyDependsOn(typeof(Avx2))]
+        [CompExactlyDependsOn(typeof(Avx512Vbmi.VL))]
+        public static Vector256<byte> ShuffleUnsafe(Vector256<byte> vector, Vector256<byte> indices)
+        {
+            if (Avx512Vbmi.VL.IsSupported)
+            {
+                return Avx512Vbmi.VL.PermuteVar32x8(vector, indices);
+            }
+
+            if (Avx2.IsSupported)
+            {
+                // check for constant value of indices where we can optimise further
+                if (RuntimeHelpers.IsKnownConstant(indices))
+                {
+                    // check for all in correct lane (or will result in 0)
+                    if (LessThanAll(Xor(BitwiseAnd(indices, Vector256.Create((byte)0x9F)), Vector256.Create(Vector128.Create((byte)0), Vector128.Create((byte)0x10))).AsSByte(), Vector256.Create((sbyte)0x10)))
+                    {
+                        return Avx2.Shuffle(vector, indices);
+                    }
+
+                    // check for all indices in first lane (or will result in 0)
+                    if (LessThanAll(BitwiseAnd(indices, Vector256.Create((byte)0x9F)).AsSByte(), Vector256.Create((sbyte)0x10)))
+                    {
+                        var laneShuffle = Avx2.Permute2x128(vector, vector, 0b00000000);
+                        return Avx2.Shuffle(laneShuffle, indices);
+                    }
+
+                    // can add other similar cases as required
+                }
+
+                // get the indices, but only keep the high bit and bits used for indexing, and xor the cross-lane bit on the high 128-bit lane part of indices
+                // we begin computing this early as it seems to take the longest to calculate (it can be done in parallel to other operations ideally)
+                var indicesXord = Avx2.And(Avx2.Xor(indices, Vector256.Create(Vector128.Create((byte)0), Vector128.Create((byte)0x10))), Vector256.Create((byte)0x9F));
+
+                // swap the low and high 128-bit lanes
+                // calculate swap before shuf1 so they can be computed in parallel
+                var swap = Avx2.Permute2x128(vector, vector, 0b00000001);
+
+                // shuffle with both the normal and swapped values
+                var shuf1 = Avx2.Shuffle(vector, indices);
```

Comment:
We prefer explicit types over `var` in runtime for cases where the type isn't obvious (e.g. as part of a ctor).